### PR TITLE
adapt go-pagerduty to simplified custom fields API

### PR DESCRIPTION
**Pull request by jedelson-pagerduty**
The Custom Fields API is being significantly simplified and "rehomed" under /incidents. This pull request updates the go client to be compatible with the new API.

New API documentation is visible at https://developer.pagerduty.com/api-reference/2131f556073c4-create-a-field
New product documentation is visible at https://support.pagerduty.com/docs/custom-fields-on-incidents

The existing public API in this library has been retained but all structs/types/functions have been marked as deprecated. All functions throw an error (except for some conversion functions which are largely left in place). Unit tests have been removed.

Once this is merged, will follow up with a separate PR to remove the structs/types/functions completely.

### CODE REVIEW

File: pagerduty/incident_custom_field.go [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+	Includes []string `url:"include,brackets,omitempty"`
+}
+
+var incidentCustomFieldsEarlyAccessHeader = RequestOptions{
+	Type:  "header",
+	Label: "X-EARLY-ACCESS",
+	Value: "flex-service-early-access",
+}
+
+// List lists existing custom fields. If a non-zero Limit is passed as an option, only a single page of results will be
```
Comment (imjaroiswebdev) [inline pagerduty/incident_custom_field.go]: Please let's just add the context aware version of the methods, it not make sense for new methods to keep both version. Please apply this everywhere applicable. Thank you 🙏🏽

**Review by imjaroiswebdev — CHANGES_REQUESTED** [review]
Hey @jedelson-pagerduty great job in the "rehoming" of *Custom Fields* to *Incidents*, however since We can consider these as new methods, please remove the support for the non context aware methods and after that, We will be ready to merge ✌🏽 💪🏽

**Comment by jedelson-pagerduty** [comment]
@imjaroiswebdev 

> please remove the support for the non context aware methods

This is done now. It looks like an unrelated test case is failing, however:

```
=== RUN   TestListAllWithLicenses
2023/05/30 12:05:05 ===== PagerDuty Cache Skipping Init =====
2023/05/30 12:05:05 ==== Getting users at offset 0
2023/05/30 12:05:05 ==== Getting users at offset 1
2023/05/30 12:05:05 ==== Getting license_allocations at offset 0
2023/05/30 12:05:05 ==== Getting license_allocations at offset 1
    user_test.go:264: returned []*pagerduty.User{(*pagerduty.User)(0xc0002f4dc0), (*pagerduty.User)(0xc0002f4c80)}; want []*pagerduty.User{(*pagerduty.User)(0xc0002f4f00), (*pagerduty.User)(0xc0002f5040)}
--- FAIL: TestListAllWithLicenses (0.00s)
```

This error, however, seems inconsistent and I can't reproduce locally. I tried switching over to https://github.com/stretchr/testify to get better failing test error messages, but of course, after doing this the test passed.

File: pagerduty/user_test.go [hunk 2]
```diff
@@ -259,10 +261,7 @@ func TestListAllWithLicenses(t *testing.T) {
 			License: &LicenseReference{ID: "P1D3XYZ", Type: "license_reference"},
 		},
 	}
-
-	if !reflect.DeepEqual(resp, want) {
-		t.Errorf("returned %#v; want %#v", resp, want)
-	}
+	assert.Equal(t, want, resp)
```
Comment (imjaroiswebdev) [inline pagerduty/user_test.go]: Yes, `testify` could help better with the output of the tests, however in this case that's not the source of the issue. Therefore, please revert this change I just submit an issue to address this bug #132

**Review by imjaroiswebdev — CHANGES_REQUESTED** [review]
Please revert changes to the Users.ListAllWithLicenses and We will be ready to merge.

**Comment by jedelson-pagerduty** [comment]
> Please revert changes to the Users.ListAllWithLicenses and We will be ready to merge.

Done

**Review by imjaroiswebdev — APPROVED** [review]
Great! Thank you @jedelson-pagerduty this looks very good to be merged 🎉 💪🏽
